stacks_queues/queue.py

```diff
@@ -13,49 +13,78 @@ def __init__(self):
         self.store = [None] * INITIAL_QUEUE_SIZE
         self.buffer_size = INITIAL_QUEUE_SIZE
         self.front = -1
-        self.rear = -1
+        self.rear = 0
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = 0
+            self.rear = 0
+        elif self.size == self.buffer_size:
+            raise QueueFullException
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.front == -1:
+            raise QueueEmptyException
+        else:
+            value = self.store[self.front]
+            self.front = (self.front + 1) % self.buffer_size
+            self.size -= 1
+        if self.front == self.rear:
+            self.front = -1
+            self.rear = -1
+        return value
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.front == -1:
+            return None
+        return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        if self.front == -1:
+            return 0
+        return self.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        return self.front == -1
 
     def __str__(self):
         """ Returns the Queue in String form like:
             [3, 4, 7]
             Starting with the front of the Queue and
             ending with the rear of the Queue.
         """
-        pass
+        result = []
+        index = self.front
+        size = self.size
+
+        for x in range(size):
+            result.append(self.store[index])
+            index = (index + 1) % self.buffer_size
+            size -= 1
```

Comment:
We generally try not to change the value we are iterating through, and it's not necessary to modify size here.
```suggestion
```